transform/transformers/cord/ecommerce_transformer.py

```diff
@@ -0,0 +1,390 @@
+
+        Note: There are negative playback codes that are in the form 'd[0-9]+'.  Each of these affects a set of questions
+        which are listed below
+        d6 - During 2018, did the business experience any of the following difficulties
+        when selling to other EU countries via a website or 'app'? - 462, 463, 464, 465 and 466
+
+        """
+        answers = {
+            "234": self.yes_no_question("234"),
+            "235": self.percentage_question("235"),
+            "257": self.yes_no_question("257"),
+            "258": self.percentage_question("258"),
+            "310": self.checkbox_question("310"),
+            "311": self.checkbox_question("311"),
+            "312": self.checkbox_question("312"),
+            "313": self.checkbox_question("313"),
+            "314": self.checkbox_question("314"),
+            "315": self.checkbox_question("315"),
+            "348": self.percentage_question("348"),
+            "349": self.percentage_question("349"),
+            "458": self.checkbox_question("458"),
+            "459": self.checkbox_question("459"),
+            "460": self.percentage_question("460"),
+            "461": self.percentage_question("461"),
+            "462": self.negative_playback_question("462", "d6"),
+            "463": self.negative_playback_question("463", "d6"),
+            "464": self.negative_playback_question("464", "d6"),
+            "465": self.negative_playback_question("465", "d6"),
+            "466": self.negative_playback_question("466", "d6"),
+        }
+
+        # These qcodes could be optional depending on routing
+        optional_qcodes = ['310', '311', '312', '458', '459', '313', '314', '315']
+        # Replace the answers for any optional ones with "0" if they're not in the submission
+        for qcode in optional_qcodes:
+            if not self.get_qcode(qcode):
+                answers[qcode] = "0"
+
+        return answers
+
+    def transform(self):
+        """Perform a transform on survey data."""
+
+        # 001 is the 'has anything changed' question that doesn't appear in eq.
+        transformed = {
+            "001": "0",
+            "500": "1" if self.response["data"].get("500") else "0",
+        }
+
+        use_of_computers = self.use_of_computers()
+        ict_specialists_and_skills = self.ict_specialists_and_skills()
+        access_and_use_of_internet = self.access_and_use_of_internet()
+        sharing_of_info_electronically_within_business = self.sharing_of_info_electronically_within_business()
+        ict_security = self.ict_security()
+        e_commerce = self.e_commerce()
+
+        logger.info("Transforming data for {}".format(self.ids.ru_ref), tx_id=self.ids.tx_id,)
+
+        sections_to_transform = [
+            transformed, ict_specialists_and_skills, access_and_use_of_internet,
+            sharing_of_info_electronically_within_business, ict_security, e_commerce,
+            use_of_computers
+        ]
```

Comment:
I started implementing it and didn't like how it looked.  Personally, I find having named functions that return pieces of it easier to comprehend rather than having one large dict.